runner/src/simulate.rs

```diff
@@ -546,7 +561,153 @@ impl ScenarioState {
+            e
+        })?;
+
+        if let Some(state) = response_json.get("state") {
+            if let Some(anchor_status) = state.get("anchorStatus") {
+                match anchor_status {
+                    serde_json::Value::String(s) => return Ok(s.clone()),
+                    serde_json::Value::Number(n) => return Ok(n.to_string()),
+                    _ => {
+                        error!(
+                            "Unexpected anchor status type in response: {}",
+                            anchor_status
+                        );
+                        bail!(
+                            "Unexpected anchor status type: expected string or number, got: {:?}",
+                            anchor_status
+                        );
+                    }
+                }
+            }
+        }
+
+        error!("Anchor status not found in the response JSON");
+        bail!("Anchor status missing in the response")
+    }
+
+    pub async fn get_set_from_redis(&self, key: &str) -> Result<Vec<String>, anyhow::Error> {
+        // Create a new Redis client
+        let client: redis::Client = get_redis_client().await?;
+        let mut connection = client.get_async_connection().await?;
+        // Get the MIDs from Redis
+        let response = redis::cmd("SMEMBERS")
+            .arg(key)
+            .query_async(&mut connection)
+            .await?;
+        // Return the MIDs
+        Ok(response)
+    }
+
+    pub async fn remove_stream_from_redis(
+        &self,
+        key: &str,
+        stream_ids: Vec<String>,
+    ) -> Result<i32, anyhow::Error> {
+        let client: redis::Client = get_redis_client().await?;
+        let mut connection = client.get_async_connection().await?;
+        let response = redis::cmd("SREM")
+            .arg(key)
+            .arg(stream_ids)
+            .query_async(&mut connection)
+            .await?;
+        Ok(response)
+    }
+
+    async fn validate_anchoring_banchmark_scenario_success(
+        &self,
+    ) -> Result<(CommandResult, Option<PeerRps>), anyhow::Error> {
+        let mut anchored_count = 0;
+        let mut pending_count = 0;
+        let mut failed_count = 0;
+        let mut not_requested_count = 0;
+        let wait_duration = Duration::from_secs(60 * 60);
+        // TODO_3164_1 : Make this a parameter, pass it in from the scenario config
+        // TODO_3164_3 : Code clean-up : Move redis calls to separate file move it out of simulate.rs
+        // TODO_3164_4 : Code clean-up : Move api call (fetch stream) to model_instance.rs?, maybe rename it
+        sleep(wait_duration).await;
+
+        // Pick a peer at random
+        let peer = self.peers.first().unwrap();
+
+        let ids = self.get_set_from_redis("anchor_mids").await?;
+        info!("Number of MIDs: {}", ids.len());
+
+        // Make an API call to get the status of request from the chosen peer
+        for stream_id in ids.clone() {
```

Comment:
Validations in Goose can be tricky... ideally we would want a metric on js-ceramic that can get us these numbers. Since most of the code depends on polling until the state changes we do not have a nice way of getting counts of streams in the pending state (or other non-terminal states). 
This should be good for now I guess, its not the best I agree.
You are right the validation is as intensive if not more intensive than the scenario